### DISCUSSION TITLE
Version 3 - Any sense of timing?

### DISCUSSION THREAD

**Discussion by dbi1** [alpinejs/alpine#669] [4. General]
Hi,

I started using AlpineJS recently, very much like how small / productive it feels. Thanks for building Alpine 👍 

I am, however, a bit worried about the CSP unsafe-eval policy requirement you mention in the readme... Saw the issue discussion, and the note that this will likely be resolved in V3.

Any pointers what kind of timeframe we might be talking about until V3 comes out?

Thank you!

**Comment by HugoDF** [comment]
I can't speak around the timeframe but the note in the README around CSP is that we **might** re-evaluate the value of a CSP-friendly build for v3 but no guarantee that it will be included.

**Comment by dbi1** [comment]
Got it, thanks @Hugo, it sounded more definitive to me in the issue itself:

"For those who follow, v3 will have a more modular approach which should allow for a csp friendly build. Unfortunately, this is not possible in v2 without introducing breaking changes." (from https://github.com/alpinejs/alpine/issues/237#issuecomment-626230965)

Would love for this to happen, but I understand if this is not a priority.

**Comment by SimoTod** [comment]
Thanks for your feedback, to clarify: v3 WILL have a modular approach and
it SHOULD allow for a csp friendly build.
It's on our radar and keen on providing a good solution but we are still
looking into options that would allow this without making Alpine too slow
and too heavy.

**Comment by dbi1** [comment]
Thank you @SimoTod, that sounds good.

Re timing of v3, is there possibly a rough sense (a few months, a year, more than a year)? Thank you again.

**Reply by ryangjchandler** [reply]
It all depends on priorities. The most important thing for us at the moment is making sure that Alpine is stable.

**Reply by dbi1** [reply]
Got it, thank you @ryangjchandler.

Will keep my fingers crossed for a not-too-distant v3 that will allow csp-friendly builds :)
